### ISSUE TITLE
Deploy to Netlify Button Fails

### ISSUE THREAD

**Issue by K-Love** [chec/commercejs-nextjs-demo-store#136]
Hi, just FYI, the Deploy to Netlify button leads to deployment failures in Netlify.

**Comment by robbieaverill** [comment]
Thanks for the heads up.

Here's the relevant output from the build log:

```
9:50:30 AM: Automatically optimizing pages...
9:50:31 AM: Warning: You have opted-out of Automatic Static Optimization due to `getInitialProps` in `pages/_app`.
9:50:31 AM: Read more: https://err.sh/next.js/opt-out-auto-static-optimization
9:50:36 AM: Error occurred prerendering page "/product/FthtQU". Read more: https://err.sh/next.js/prerender-error:
9:50:36 AM: TypeError: Cannot read property 'replace' of null
9:50:36 AM: Error occurred prerendering page "/product/vDTUAE". Read more: https://err.sh/next.js/prerender-error:
9:50:36 AM: TypeError: Cannot read property 'replace' of null
9:50:37 AM: Error occurred prerendering page "/product/c6YdQ5". Read more: https://err.sh/next.js/prerender-error:
9:50:37 AM: TypeError: Cannot read property 'replace' of null
9:50:37 AM: Error occurred prerendering page "/product/qDsxgz". Read more: https://err.sh/next.js/prerender-error:
9:50:37 AM: TypeError: Cannot read property 'replace' of null
9:50:37 AM: Error occurred prerendering page "/product/Nzp0P3". Read more: https://err.sh/next.js/prerender-error:
9:50:37 AM: TypeError: Cannot read property 'replace' of null
9:50:37 AM: Error occurred prerendering page "/product/cmJAKu". Read more: https://err.sh/next.js/prerender-error:
9:50:37 AM: TypeError: Cannot read property 'replace' of null
9:50:38 AM: Error occurred prerendering page "/product/GVANEG". Read more: https://err.sh/next.js/prerender-error:
9:50:38 AM: TypeError: Cannot read property 'replace' of null
9:50:38 AM: Error occurred prerendering page "/product/8OxIgO". Read more: https://err.sh/next.js/prerender-error:
9:50:38 AM: TypeError: Cannot read property 'replace' of null
9:50:39 AM: Error occurred prerendering page "/product/s0eyA9". Read more: https://err.sh/next.js/prerender-error:
9:50:39 AM: TypeError: Cannot read property 'replace' of null
9:50:39 AM: Error occurred prerendering page "/product/gWIage". Read more: https://err.sh/next.js/prerender-error:
9:50:39 AM: TypeError: Cannot read property 'replace' of null
9:50:39 AM: Error occurred prerendering page "/product/rZr3v9". Read more: https://err.sh/next.js/prerender-error:
9:50:39 AM: TypeError: Cannot read property 'replace' of null
9:50:40 AM: Error occurred prerendering page "/product/vodR2O". Read more: https://err.sh/next.js/prerender-error:
9:50:40 AM: TypeError: Cannot read property 'replace' of null
9:50:41 AM: Error occurred prerendering page "/product/f92BLp". Read more: https://err.sh/next.js/prerender-error:
9:50:41 AM: TypeError: Cannot read property 'replace' of null
9:50:41 AM: Error occurred prerendering page "/product/cmZIcu". Read more: https://err.sh/next.js/prerender-error:
9:50:41 AM: TypeError: Cannot read property 'replace' of null
9:50:41 AM: Error occurred prerendering page "/product/7epcxD". Read more: https://err.sh/next.js/prerender-error:
9:50:41 AM: TypeError: Cannot read property 'replace' of null
9:50:41 AM: Error occurred prerendering page "/product/tXJKZr". Read more: https://err.sh/next.js/prerender-error:
9:50:41 AM: TypeError: Cannot read property 'replace' of null
9:50:43 AM: Error occurred prerendering page "/product/vVXyBr". Read more: https://err.sh/next.js/prerender-error:
9:50:43 AM: TypeError: Cannot read property 'replace' of null
9:50:43 AM: Error occurred prerendering page "/product/Qfsb9H". Read more: https://err.sh/next.js/prerender-error:
9:50:43 AM: TypeError: Cannot read property 'replace' of null
9:50:43 AM: Error occurred prerendering page "/product/KYwEZj". Read more: https://err.sh/next.js/prerender-error:
9:50:43 AM: TypeError: Cannot read property 'replace' of null
9:50:43 AM: Error occurred prerendering page "/product/7WBf7Z". Read more: https://err.sh/next.js/prerender-error:
9:50:43 AM: TypeError: Cannot read property 'replace' of null
9:50:43 AM: > Build error occurred
9:50:43 AM: Error: Export encountered errors
9:50:43 AM:     at _default (/opt/build/repo/node_modules/next/dist/export/index.js:19:1086)
9:50:43 AM:     at processTicksAndRejections (internal/process/task_queues.js:97:5)
9:50:43 AM:     at async build (/opt/build/repo/node_modules/next/dist/build/index.js:35:218)
9:50:43 AM: error Command failed with exit code 1.
```

**Comment by jaepass** [comment]
Hi Kevin, did you use the provided [public key](https://github.com/chec/commercejs-nextjs-demo-store#one-click-deploy-with-netlify-recommended) mentioned in the README when this error happened or your own?
<img width="937" alt="Screen Shot 2020-10-26 at 4 07 56 PM" src="https://user-images.githubusercontent.com/48780927/97237959-7aaaf900-17a5-11eb-945e-75ea273fb244.png">

**Comment by K-Love** [comment]
Hi, 

I used my own key. 


Kevin Love
Founder

Web | Email | Blog

**Comment by robbieaverill** [comment]
It turns out that product descriptions are not always being returned as strings. I've added some truthy checks to our string manipulation on this property, fix is at #137
